src/org/daisy/dotify/hyphenator/impl/JHyphenator.java

```diff
@@ -0,0 +1,55 @@
+package org.daisy.dotify.hyphenator.impl;
+
+import ch.sbs.jhyphen.Hyphenator;
+import org.daisy.dotify.api.hyphenator.HyphenatorConfigurationException;
+
+import java.io.File;
+import java.util.PropertyResourceBundle;
+import java.util.ResourceBundle;
+
+class JHyphenator extends AbstractHyphenator {
+    private final static char SHY = '\u00AD';
+    private final static char ZWSP = '\u200B';
+    private Hyphenator instance;
+
+	JHyphenator(String locale) throws HyphenatorConfigurationException {
+	    try {
+            ResourceBundle p = PropertyResourceBundle.getBundle("org/daisy/dotify/hyphenator/impl/JHyphenator");
+            instance = new Hyphenator(new File("/usr/share/hyphen/", p.getString(locale)));
+        } catch (Exception e) {
+	        throw new JHyphenatorConfigurationException(e);
+        }
+	}
+
+	@Override
+	public String hyphenate(String phrase) {
+		return ensureStartAndEnd(instance.hyphenate(phrase, SHY, ZWSP));
```

Comment:
I believe this will throw an exception if the word contains non-standard hyphenation points.